runtime/compiler/control/CompilationThread.cpp

```diff
@@ -7712,8 +7712,9 @@ TR::CompilationInfoPerThreadBase::preCompilationTasks(J9VMThread * vmThread,
       if ((_compInfo.getPersistentInfo()->getRemoteCompilationMode() == JITServer::CLIENT) &&
           TR::Options::canJITCompile())
          {
-         bool forcedLocal = false;
-         bool doLocalCompilation = entry->isAotLoad() || cannotDoRemoteCompilation || preferLocalComp(entry);
+	 TR_ResolvedMethod *resolvedMethod = fe->createResolvedMethod(&trMemory, (TR_OpaqueMethodBlock *)method);
```

Comment:
This is not the right place to check for filters because it misses AOT compilations which happen under this `if` statement:
```
  if (canDoRelocatableCompile)
```
Probably the best place is after this statements:
```
#if defined(J9VM_OPT_JITSERVER)
   bool cannotDoRemoteCompilation = cannotPerformRemoteComp(
#if defined(J9VM_OPT_CRIU_SUPPORT)
      vmThread
#endif /* defined(J9VM_OPT_CRIU_SUPPORT) */
   );
#endif /* defined(J9VM_OPT_JITSERVER) */

   if (!entry->isAotLoad()) // We don't/can't relocate this method
      {
```
We know that AOT loads are never sent to the server, so this path captures all the remote compilations.
You just check the filters and adjust `cannotDoRemoteCompilation` accordingly.

Reply:
@mpirvu Or Can the code be added as an additional check in the method `preferLocalComp(..`

After this ?
```
bool
TR::CompilationInfoPerThreadBase::preferLocalComp(const TR_MethodToBeCompiled *entry)
   {
   if (_compInfo.getPersistentInfo()->isLocalSyncCompiles() &&
       (entry->_optimizationPlan->getOptLevel() <= cold) &&
       !entry->_async)
      {
      return true;
      }
```
Will have to pass the `TR_ResolvedMethod` as an additional argument to the method, as I could not find if I can get the signature from a `J9Method` object.